File: android-compiler/src/main/java/com/pranav/lib_android/code/disassembler/EclipseDisassembler.java
package com.pranav.lib_android.code.disassembler;

import kotlin.io.FilesKt;

import org.eclipse.jdt.internal.core.util.Disassembler;

import java.io.File;
import java.io.IOException;

public class EclipseDisassembler {

    final byte[] classFileBytes;

    public EclipseDisassembler(String classFile) throws IOException {
        classFileBytes = FilesKt.readBytes(new File(classFile));
    }

    public String disassemble() throws Throwable {
        return new Disassembler().disassemble(classFileBytes, System.lineSeparator());
    }
}

File: android-compiler/src/main/java/org/cosmic/ide/android/code/disassembler/EclipseDisassembler.kt
package org.cosmic.ide.android.code.disassembler

import org.eclipse.jdt.core.util.ClassFileBytesDisassembler
import org.eclipse.jdt.internal.core.util.Disassembler
import java.nio.file.Files
import java.nio.file.Paths

class EclipseDisassembler(filePath: String) {

    private var classFileBytes: ByteArray

    init {
        classFileBytes = Files.readAllBytes(Paths.get(filePath))
    }

    @Throws(Throwable::class)
    fun disassemble(): String {
        return Disassembler().disassemble(classFileBytes, System.lineSeparator(), ClassFileBytesDisassembler.SYSTEM)
    }
}
